core/src/io/Datawriter.cpp

```diff
@@ -54,10 +58,11 @@ namespace IO
 		if (readability_toggle) header = separator + line + separator;
 		else header = line;
 		if (!readability_toggle) std::replace( header.begin(), header.end(), '|', ' ');
-		String_to_File(header, fileName);
```

Comment:
All the formatting changes (indentation, where you break lines etc... and now this kind of change... You do too much of this at once! This PR is very hard to keep track of, as you pile a lot of these aesthetic changes on top of the refactoring. Please try to keep this to a minimum and rather create another PR later, where you only do clean-up.